### PR TITLE
Refactor file copying tasks to remove Gulp

### PR DESCRIPTION
**Pull request by colinrotherham**
This PR removes Gulp from the `generateFixtures()` `generateMacroOptions()` tasks to complete:

* https://github.com/alphagov/govuk-frontend/issues/2711

### CODE REVIEW

**Comment by colinrotherham** [comment]
@dependabot rebase

**Review by domoscargin — APPROVED** [review]
Looks good to me!

The only diff in the outputted JSON files is that they don't have EOLs now. The version file in dist is exactly the same, and I like that we've dropped a few dependencies.

Nice stuff.

Did some unscientific bulk testing of `npm run build:package` and there's nothing in it for speed.

I don't think the resulting code is much lengthier or less understandable compared to the gulp-based approach.

File: tasks/configs.mjs [hunk 1]
```diff
@@ -10,13 +10,18 @@ import { parse, join } from 'path'
...
-  const { default: configFn } = await import(join(srcPath, modulePath))
+  // Write to destination
+  return files.write(modulePath, {
+    ...options,
 
-  // Write JSON config file
-  return writeFile(configPath, JSON.stringify(await configFn(), null, 2) + EOL)
+    // Format config as JSON
+    async fileContents () {
+      return JSON.stringify(await configFn(), undefined, 2)
```
[the remark this answers is not in the document]
Comment (colinrotherham) [reply tasks/configs.mjs:22]: Ah it's just been [moved it to `files.write()`](https://github.com/alphagov/govuk-frontend/blob/ungulp-json/tasks/files.mjs#L50) instead

**Comment by colinrotherham** [comment]
> The only diff in the outputted JSON files is that they don't have EOLs now. The version file in dist is exactly the same, and I like that we've dropped a few dependencies.

Thanks @domoscargin

Can you double check your diff direction? After this PR merges we should see trailing newline characters (as a consistency improvement) in `fixtures.json` and `macro-options.json` where they were missed previously

Although to reduce the diff size I've kept the 4-space indent (rather than our 2-space default)

**Pull request merged**